sunpy/map/sources/gong.py

```diff
@@ -61,3 +71,56 @@ def spatial_units(self):
     @property
     def observer_coordinate(self):
         return get_earth(self.date)
+
+
+class GONGHalphaMap(GenericMap):
+    """
+    GONG H-Alpha Map.
+
+    The Global Oscillation Network Group (GONG) operates a six-station network of H-alpha
+    imagers located around the Earth that observe the Sun nearly continuously.
+
+    References
+    ----------
+    * `GONG H-Alpha Page <https://nso.edu/data/nisp-data/h-alpha/>`_
+    * `GONG H-Alpha Observation Details <https://nispdata.nso.edu/webProdDesc2/presenter.php?file=halpha_fulldisk_images_overview.html&echoExact=0&name=Overview%20:%20GONG%20H-alpha%20Full-disk%20Images>`_
+    * `GONG Header Keywords <https://gong.nso.edu/data/HEADER_KEY.html>`_
+    * `DOI:/10.25668/as28-7p13 <https://doi.org/10.25668/as28-7p13>`_
+    """
+    def __init__(self, data, header, **kwargs):
+        header['WCSNAME'] = 'Helioprojective-Cartesian'
+        header['CTYPE1'] = 'HPLN-TAN'
+        header['CTYPE2'] = 'HPLT-TAN'
+        super().__init__(data, header, **kwargs)
+
+    @classmethod
+    def is_datasource_for(cls, data, header, **kwargs):
+        return (str(header.get('TELESCOP', '')).endswith('GONG') and
+                str(header.get('IMTYPE', '')).startswith('H-ALPHA'))
+
+    @property
+    def date(self):
+        return Time(self.meta.get('date-obs'))
+
+    @property
+    def scale(self):
+        return SpatialPair((self.meta['SOLAR-R'] * u.arcsec) / (self.meta['FNDLMBMI'] * u.pixel),
+                           (self.meta['SOLAR-R'] * u.arcsec) / (self.meta['FNDLMBMA'] * u.pixel))
+
+    @property
+    def nickname(self):
+        site = _SITE_NAMES.get(self.meta.get("sitename", ""), "UNKNOWN")
+        return f'{self.observatory}, {site}'
+
+    @property
+    def spatial_units(self):
+        return SpatialPair(u.deg, u.deg)
+
+    @property
+    def earth_location(self):
+        """Location of the observatory on Earth"""
+        return EarthLocation.from_geodetic(lat=self.meta['site-lat'] * u.deg, lon=self.meta['site-lon'] * u.deg)
+
+    @property
+    def observer_coordinate(self):
+        return SkyCoord(self.earth_location.itrs).heliographic_stonyhurst
```

Comment:
Do we have a test for this?

Reply:
No but I can add one but there is a test for `self.earth_location` and if this were to break it would be in `SkyCoord` or `EarthLocation`